drivers/FileSR.py

```diff
@@ -913,6 +917,31 @@ def extractUuid(path):
         return uuid
     extractUuid = staticmethod(extractUuid)
 
+    def generate_config(self, sr_uuid, vdi_uuid):
+        util.SMlog("FileVDI.generate_config")
+        if not util.pathexists(self.path):
+                raise xs_errors.XenError('VDIUnavailable')
+        resp = {}
+        resp['device_config'] = self.sr.dconf
+        resp['sr_uuid'] = sr_uuid
+        resp['vdi_uuid'] = vdi_uuid
+        resp['command'] = 'vdi_attach_from_config'
+        # Return the 'config' encoded within a normal XMLRPC response so that
+        # we can use the regular response/error parsing code.
+        config = xmlrpclib.dumps(tuple([resp]), "vdi_attach_from_config")
+        return xmlrpclib.dumps((config,), "", True)
+
+    def attach_from_config(self, sr_uuid, vdi_uuid):
+        """Used for HA State-file only. Will not just attach the VDI but
+        also start a tapdisk on the file"""
+        util.SMlog("FileVDI.attach_from_config")
+        try:
+           if not util.pathexists(self.sr.path):
+                self.sr.attach(sr_uuid)
+        except:
+            util.logException("FileVDI.attach_from_config")
+            raise xs_errors.XenError('SRUnavailable', \
+                        opterr='Unable to attach from config')
```

Comment:
Sorry to nitpick but even if it does not seem so we are striving to adhere tp PEP-08
https://www.python.org/dev/peps/pep-0008/#indentation

You can either do this
```python
raise xs_errors.XenError(
    'SRUnavailable',
    opterr='Unable to attach from config'
)
```
this
```python
raise xs_errors.XenError(
    'SRUnavailable',
    opterr='Unable to attach from config')
```
or this
```python
raise xs_errors.XenError('SRUnavailable',
                         opterr='Unable to attach from config')
```